google/cloud/forseti/scanner/scanner.py

```diff
@@ -39,11 +97,14 @@ def run(model_name=None, progress_queue=None, service_config=None):
     violation_access = scanner_dao.define_violation(service_config.engine)
     service_config.violation_access = violation_access
 
+    scanner_index_id = init_scanner_index(service_config)
     runnable_scanners = scanner_builder.ScannerBuilder(
         global_configs, scanner_configs, service_config, model_name,
         None).build()
 
     # pylint: disable=bare-except
+    succeeded = []
```

Comment:
Instead of having 2 lists, you can try to have 1 list `run_states`.

Then you can just append `True` on success, or `False` on failure to `run_states`.  Then, you can go ```if all(run_states)```, or ```if any(run_states)``` to evaluate for the actual states.

This logic table should help.
https://stackoverflow.com/a/19389957

See my next comment below, on putting this into a helper method, and simplify `mark_scanner_index_complete()`